src/CodeCheckers/FunctionChecker.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace CodeCheckers;
+
+
+
+use Nette\Utils\Strings;
+
+class FunctionChecker
+{
+
+	public static function createChecker()
+	{
+		return function ($checker, $s) {
+			if (!$checker->is('php,phpt')) {
+				return;
+			}
+			if (Strings::match($s, '~interface ~')) {
+				return;
+			}
+			if (!Strings::match($s, '~^(abstract\s+)?class~m')) {
+				return;
+			}
+
+			if ($matchAll = Strings::matchAll($s, '~^([\s])+function(.*)~m')) {
```

Comment:
>>>
mzk [13:10] 
@fprochazka: u čeho by failnula?

mzk [13:10]
je nějaký testcase, který to odhalí špatně? Nebo který ignoruje?

fprochazka [13:11] 
@mzk je to proste spatne napsany, spatnej  pristup... to je jak kdybys chtěl zkontrolovat jestli ma auto dost oleje a koukal do nej pres vyfuk

mzk [13:11] 
třeba tam pastni string, který si myslíš, že neprojde

fprochazka [13:12] 
nechceme reseni ktere "nahodou prochazi", chceme reseni ktere se chova co nejvic korektně (edited)

mzk [13:12] 
btw, operaci břicha taky dělají přes prdel :simple_smile:

mzk [13:12]
když už jsme u těch přirovnání

fprochazka [13:13] 
``` public function
 function_tralala() {}
```

fprochazka [13:13]
druhej bude hlasit ze nema viditelnost

fprochazka [13:13]
pritom viditelnost ma

fprochazka [13:14]
vim, ze to tak nepiseme, ale ta tvoje kontrola je proste strasne krehka a nespolehliva

mzk [13:14] 
naopak je správně

mzk [13:14]
vyhodí error, že to není dle CS

fprochazka [13:14] 
a nechci aby nam vynucovalo pravidla nad kodem neco, co je krehke a nespolehlive

mzk [13:14] 
sice nevalidní error message

mzk [13:14]
ale i tak to pomůže..

fprochazka [13:14] 
presne tak, napise to picovinu

mzk [13:14] 
přesně tak, pomůže to odhalit nevalidní cs

mzk [13:15]
takže co kdybych upravil hlášku z "Missing visibility: %s” na "Missing visibility line %s”